### PR TITLE
[Development] Add new predicate for multi-agent task

### PR DESCRIPTION
**Pull request by Tradewindycc**
## Motivation and Context
add is_detected and any_at predicate for multi-agent task, modify the DroneRobot params

## How Has This Been Tested
Set the goal in [PDDL config yaml](https://github.com/SgtVincent/habitat-lab/blob/0f96aafb5f9936e7b877ef2245bf0c4159684bb0/habitat-lab/habitat/config/benchmark/multi_agent/pddl/zxz_spot_drone_per.yaml) like` is_detected(any_targets|0)` and `any_at(any_targets|0)`

## Types of changes

- **\[Docs change\]** Addition or changes to the documentation
- **\[Refactoring\]** Large changes to the code that improve its functionality or performance
- **\[Dependency Upgrade\]** Upgrades one or several dependencies in habitat
- **\[Bug Fix\]** (non-breaking change which fixes an issue)
- **\[Development\]** A pull request that add new features to the [habitat-lab](/habitat-lab) task and environment codebase. Development Pull Requests must be small (less that 500 lines of code change), have unit testing, very extensive documentation and examples. These are typically new tasks, environments, sensors, etc... The review process for these Pull Request is longer because these changes will be maintained by our core team of developers, so make sure your changes are easy to understand!
- **\[Experiment\]** A pull request that add new features to the [habitat-baselines](/habitat-baselines/) training codebase. Experiments Pull Requests can be any size, must have smoke/integration tests and be isolated from the rest of the code. Your code additions should not rely on other habitat-baselines code. This is to avoid dependencies between different parts of habitat-baselines. You must also include a README that will document how to use your new feature or trainer. **You** will be the maintainer of this code, if the code becomes stale or is not supported often enough, we will eventually remove it.

## Checklist

- [x] My code follows the code style of this project.
- [x] I have updated the documentation if required.
- [x] I have read the [**CONTRIBUTING**](/CONTRIBUTING.md) document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [x] I have added tests to cover my changes if required.

### CODE REVIEW

File: habitat-baselines/habitat_baselines/rl/multi_agent/habitat_mas_evaluator.py
Comment (Zhouxunzhe) [inline habitat-baselines/habitat_baselines/rl/multi_agent/habitat_mas_evaluator.py:1]: Why do you change this file back to the former edition?

File: habitat-lab/habitat/config/habitat/task/actions/drone_oracle_nav_action.yaml
Comment (Zhouxunzhe) [inline habitat-lab/habitat/config/habitat/task/actions/drone_oracle_nav_action.yaml:1]: What is this change for?

File: habitat-lab/habitat/tasks/rearrange/multi_task/pddl_sim_state.py [hunk 1]
```diff
@@ -381,6 +391,43 @@ def type_matches(entity, match_names):
...
+        obs = [abs_obj_id]
+        num_robots = len(sim_info.robot_ids)
+        for i in range(num_robots):
+            if f"agent_{i}_detected_objects" in sim_info.env.sensor_suite.sensors:
+                obs = sim_info.env.sensor_suite.get(f"agent_{i}_detected_objects").get_observation(observations)
+            elif "detected_objects" in sim_info.env.sensor_suite.sensors:
+                obs = sim_info.env.sensor_suite.get("detected_objects").get_observation(observations)
+            if abs_obj_id in obs:
+                return True
+
```
Comment (Zhouxunzhe) [inline habitat-lab/habitat/tasks/rearrange/multi_task/pddl_sim_state.py:411]: Do we need to set some thresh for object detected sensor?

File: habitat-lab/habitat/tasks/rearrange/actions/habitat_mas_actions.py [hunk 2]
```diff
@@ -256,10 +258,12 @@ def action_space(self):
 
     def reset(self, *args, **kwargs):
         super().reset(*args, **kwargs)
+        self.prev_nav_done = False
+        self.skill_done = False
         if self._task._episode_id != self._prev_ep_id:
             self._targets = {}
             self._prev_ep_id = self._task._episode_id
-            self.skill_done = False
+
```
Comment (Zhouxunzhe) [inline habitat-lab/habitat/tasks/rearrange/actions/habitat_mas_actions.py:266]: Why do we need to move `self.skill_done` ahead?

**Review by Zhouxunzhe — COMMENTED** [review]
Please answer the questions at your convenience

File: habitat-lab/habitat/tasks/rearrange/actions/habitat_mas_actions.py [hunk 3]
```diff
@@ -256,10 +258,12 @@ def action_space(self):
 
     def reset(self, *args, **kwargs):
         super().reset(*args, **kwargs)
+        self.prev_nav_done = False
+        self.skill_done = False
         if self._task._episode_id != self._prev_ep_id:
             self._targets = {}
             self._prev_ep_id = self._task._episode_id
-            self.skill_done = False
```
Comment (Zhouxunzhe) [inline habitat-lab/habitat/tasks/rearrange/actions/habitat_mas_actions.py:262]: Why do we need to move self.skill_done ahead?

File: habitat-lab/habitat/config/habitat/task/actions/drone_oracle_nav_action.yaml [hunk 4]
```diff
@@ -22,6 +22,7 @@ turn_thresh: 0.5
 
 # For re-computing navmesh for different bases 
 base_type: "flying"
-agent_height: 1.5
-agent_max_climb: 1.0
+agent_height: 1.0
+agent_radius: 0.2
+agent_max_climb: 1.5
```
Comment (Zhouxunzhe) [inline habitat-lab/habitat/config/habitat/task/actions/drone_oracle_nav_action.yaml:27]: What is this change for?

File: habitat-baselines/habitat_baselines/rl/multi_agent/habitat_mas_evaluator.py [hunk 5]
```diff
@@ -196,13 +196,8 @@ def evaluate_agent(
...
             else:
-                # step_data = [a.item() for a in action_data.env_actions.cpu()]
-                env_actions = action_data.env_actions.cpu()
-                if agent.num_total_agents < 2:
-                    step_data = [a.item() for a in env_actions]
-                else:
-                    # step_data = [item.item() for a in env_actions for item in a]
-                    step_data = [a for a in env_actions]
+                step_data = [a.item() for a in action_data.env_actions.cpu()]
+
```
Comment (Zhouxunzhe) [inline habitat-baselines/habitat_baselines/rl/multi_agent/habitat_mas_evaluator.py:200]: Why do you change this file back to the former edition?

File: habitat-lab/habitat/tasks/rearrange/multi_task/pddl_sim_state.py (hunk 1 above)
Comment (Zhouxunzhe) [inline habitat-lab/habitat/tasks/rearrange/multi_task/pddl_sim_state.py:411]: Do we need to set some thresh for object detected sensor?

**Review by Zhouxunzhe — COMMENTED** [review]
Please respond the questions at your convenience.

File: habitat-lab/habitat/config/habitat/task/actions/drone_oracle_nav_action.yaml (hunk 4 above)
Comment (Tradewindycc) [reply habitat-lab/habitat/config/habitat/task/actions/drone_oracle_nav_action.yaml:27]: For higher success rate of drone detection. 1.5 is too high for drone to detect object on some receptacle, which could be blocked by the recep.

File: habitat-lab/habitat/tasks/rearrange/actions/habitat_mas_actions.py (hunk 3 above)
Comment (Tradewindycc) [reply habitat-lab/habitat/tasks/rearrange/actions/habitat_mas_actions.py:262]: `self.skill_done` only reset when it comes to a new episode, we'd better reset it once the OracleNavAction is reset.

File: habitat-baselines/habitat_baselines/rl/multi_agent/habitat_mas_evaluator.py (hunk 5 above)
Comment (Tradewindycc) [reply habitat-baselines/habitat_baselines/rl/multi_agent/habitat_mas_evaluator.py:200]: The former edition is designed for **Mobility** task based on `ObjectNav`, I forgot to revert it anyway.

File: habitat-lab/habitat/tasks/rearrange/multi_task/pddl_sim_state.py (hunk 1 above)
Comment (Tradewindycc) [reply habitat-lab/habitat/tasks/rearrange/multi_task/pddl_sim_state.py:411]: Thanks for your advice! This snippet is for sensor observation check, maybe we could add some thresh for detected sensor [here](https://github.com/SgtVincent/habitat-lab/blob/dff71f2a796fb84a47db208f5dfbd5b642813cf7/habitat-lab/habitat/tasks/rearrange/rearrange_sensors.py#L1386)

File: habitat-lab/habitat/config/habitat/task/actions/drone_oracle_nav_action.yaml (hunk 4 above)
Comment (Zhouxunzhe) [reply habitat-lab/habitat/config/habitat/task/actions/drone_oracle_nav_action.yaml:27]: Sounds reasonable!

**Comment by Zhouxunzhe** [comment]
LGTM!

**Pull request merged**